rpcs3/Emu/Cell/PPUTranslator.cpp

```diff
@@ -345,16 +345,23 @@ void PPUTranslator::CallFunction(u64 target, Value* indirect)
 
 	if (!indirect)
 	{
-		if ((!m_reloc && target < 0x10000) || target >= 0x100000000u - 0x10000)
+		const u64 base = m_reloc ? m_reloc->addr : 0;
+		const u32 caddr = m_info.segs[0].addr;
+		const u32 cend = caddr + m_info.segs[0].size - 1;
+		const u64 _target = target + base;
+
+		if (_target >= caddr && _target <= cend)
 		{
-			Trap();
-			return;
+			callee = m_module->getOrInsertFunction(fmt::format("__0x%x", target), type);
+			cast<Function>(callee.getCallee())->setCallingConv(CallingConv::GHC);
+		}
+		else
```

Comment:
Needs some kind of bounds checking against stupidly written patches.

Reply:
Does it? PPU interpreter behaves the same here.